helpers/ContentLoader.php

```diff
@@ -185,6 +187,8 @@ public function fetch($source) {
             
             \F3::get('logger')->log('Memory usage: '.memory_get_usage(), \DEBUG);
             \F3::get('logger')->log('Memory peak usage: '.memory_get_peak_usage(), \DEBUG);
+            
+            $newPostAdded = TRUE;
```

Comment:
Instead of an boolean ```$newPostAdded``` parameter you could use something similar to ```$lastEntryDate = max($lastEntryDate, $itemDate->getTimestamp());```. Lines 61-64 would then be ```$lastEntryDate = $source['lastentry']; $this->updateSource($source, null);```.